Bumps eth-lattice-keyring to v0.10.0

alex-miller-0 opened the pull request:
Sorry for all the PRs. I got confused about the diffs. This branch contains GridPlus changes + related lavamost changes. We also needed to run lavamoat a second time to account for `metamask/develop` changes because our tests were breaking in #15244. We included @danjm's `metamask/develop`-based lavamoat changes from https://github.com/MetaMask/metamask-extension/tree/gridplus-update-elk-v0.10.0 and squashed everything into a single commit.

## Explanation
GridPlus has shipped a new version of firmware for the Lattice1. This new version of the firmware includes support for automatic decoding of ABI data, which is then displayed for users on their device.

In order to accomplish this for MetaMask users, we will need to bump the version of our `eth-lattice-keyring`, which itself updates `gridplus-sdk`.

* `eth-lattice-keyring` changes: https://github.com/GridPlus/eth-lattice-keyring/compare/v0.7.3...v0.10.0
* `gridplus-sdk` changes (which includes a codebase rewrite): https://github.com/GridPlus/gridplus-sdk/compare/v1.2.3...v2.2.0

alex-miller-0 commented:
### Build issues

We were having issues with the build tests failing against our `eth-lattice-keyring` version bump commit [ce0a261](https://github.com/MetaMask/metamask-extension/pull/15261/commits/ce0a261c12484b6c7a5c5f01012f0768c9e0c406) (for example, [this one](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/25053/workflows/3918f9fa-9bf2-40c5-b5cc-c829a2ab5030/jobs/643605)). Note that this commit had lavamoat changes from `yarn lavamoat:auto` included in it.

We decided to run `yarn lavamoat:auto` a *second* time, which appears to have added the babel things that the builds were complaining about. So we put that into an additional commit: [6f9981f](https://github.com/MetaMask/metamask-extension/pull/15261/commits/6f9981f600dd1ce1fd0cef828e7136bdf76117fa).

Unfortunately, this appears to *still* be failing. Really at a loss as to why.

### Additional changes

he only additional diff we have is an updated `superagent` version (v3.8->v7). We previously tried to lock on the same version that MetaMask uses to avoid additional package-lock diffs, but it was updated because of a deprecation notice:

> npm WARN deprecated superagent@3.8.3: Please upgrade to v7.0.2+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>.

We are happy to roll back that superagent version and release a v2.2.3 of the SDK if you would like.

alex-miller-0 commented:
Making note that @danjm tried a different approach, yielding a much smaller diff (https://github.com/MetaMask/metamask-extension/commit/497fff0837c11f9dcbfc562bb9df294328dd1822), which failed in a different test (but passed the build): https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/25075/workflows/6e9aa51c-82d0-429e-a1c0-32d8d1ca0888/jobs/644483

The approach was:

1. Branch off `develop` and update the keyring module in package.json
2. `rm -rf node_modules && yarn setup && yarn lavamoat:auto`

I did not remove node modules for this branch, which seems to be why my diff is so much larger.

alex-miller-0 commented:
Just force pushed a single commit which does *not* contain lavamoat changes related to MetaMask's `develop` branch. It was created through the same methodology Dan used:

```
rm -rf node_modules
yarn setup
yarn yarn-deduplicate
yarn allow-scripts auto
yarn lavamoat:auto
```

My diff is much larger, but I am on Mac OS and Dan is on Linux so I suppose this is to be expected.

The goal is to hit the same failed test that Dan hit, i.e. move past the build step that was failing before 🤞

alex-miller-0 commented:
Tests now passing after https://github.com/MetaMask/metamask-extension/pull/15297

jpuri commented:
@alex-miller-0 : can you plz update the PR with latest from develop, it should fix the build issues.

alex-miller-0 committed:
Updates `eth-lattice-keyring` to v0.10.0
This is mainly associated with an update in GridPlus SDK and enables
better strategies for fetching calldata decoder data.
`eth-lattice-keyring` changes:
GridPlus/eth-lattice-keyring@v0.7.3...v0.10.0
`gridplus-sdk` changes (which includes a codebase rewrite):
GridPlus/gridplus-sdk@v1.2.3...v2.2.2

alex-miller-0 commented:
Thanks @jpuri - they passed!
